modules/sdk-coin-ada/src/lib/transactionBuilder.ts

```diff
@@ -161,11 +161,70 @@ export abstract class TransactionBuilder extends BaseTransactionBuilder {
       // add extra output for the change
       if (this._changeAddress && this._senderBalance) {
         const changeAddress = CardanoWasm.Address.from_bech32(this._changeAddress);
-        const mockChange = CardanoWasm.TransactionOutput.new(
-          changeAddress,
-          CardanoWasm.Value.new(CardanoWasm.BigNum.from_str(this._senderBalance))
-        );
-        outputs.add(mockChange);
+        const utxoBalance = CardanoWasm.BigNum.from_str(this._senderBalance);
+
+        const adjustment = BigNum.from_str('2000000');
+        let change = utxoBalance.checked_sub(this._fee).checked_sub(totalAmountToSend);
+        if (this._type === TransactionType.StakingActivate) {
+          change = change.checked_sub(adjustment);
+        } else if (this._type === TransactionType.StakingDeactivate) {
+          change = change.checked_add(adjustment);
+        } else if (this._type === TransactionType.StakingWithdraw || this._type === TransactionType.StakingClaim) {
+          this._withdrawals.forEach((withdrawal: Withdrawal) => {
+            change = change.checked_add(CardanoWasm.BigNum.from_str(withdrawal.value));
+          });
+        }
+
+        // If totalAmountToSend is 0, its consolidation
+        if (totalAmountToSend.to_str() == '0') {
+          // support for multi-asset consolidation
+          if (this._multiAssets !== undefined) {
+            const totalNumberOfAssets = CardanoWasm.BigNum.from_str(this._multiAssets.length.toString());
+            const minAmountNeededForOneAssetOutput = CardanoWasm.BigNum.from_str('1500000');
```

Comment:
nit: can we put this in a top level variable in case it needs to be changed in the future?